_source/_posts/2021-09-16-spring-native-okta-starter.adoc

```diff
@@ -0,0 +1,179 @@
+
+You can also try it out yourself with the https://cli.okta.com[Okta CLI]. Install it and follow along below for a quick Okta + Spring Native example.
+
+== Create a Native Spring Boot App
+
+To create a secure Spring Boot app with Okta, run `okta start spring-boot`. You'll need to verify your email and set a password as part of this.
+
+TIP: If you already have an Okta account, you can run `okta login` first.
+
+This command will download our https://github.com/okta-samples/okta-spring-boot-sample[Okta Spring Boot sample], register your app on Okta, and configure it by adding your Okta settings to `src/main/resources/application.properties`.
+
+Add `@NativeHint(options = "--enable-https")` to the main `Application` class as specified above.
+
+Next, edit your `pom.xml` and add the Spring Native version and classifier to the `<properties>` section:
+
+[source,xml]
+----
+<spring-native.version>0.10.3</spring-native.version>
+<repackage.classifier/>
+----
+
+Then, replace the `<build>` section with the XML below:
+
+[source,xml]
+----
+<build>
+    <defaultGoal>spring-boot:run</defaultGoal>
+    <plugins>
+        <plugin>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-maven-plugin</artifactId>
+            <configuration>
+                <classifier>${repackage.classifier}</classifier>
+                <image>
+                    <builder>paketobuildpacks/builder:tiny</builder>
+                    <env>
+                        <BP_NATIVE_IMAGE>true</BP_NATIVE_IMAGE>
+                    </env>
+                </image>
+            </configuration>
+        </plugin>
+        <plugin>
+            <groupId>org.springframework.experimental</groupId>
+            <artifactId>spring-aot-maven-plugin</artifactId>
+            <version>${spring-native.version}</version>
+            <executions>
+                <execution>
+                    <id>test-generate</id>
+                    <goals>
+                        <goal>test-generate</goal>
+                    </goals>
+                </execution>
+                <execution>
+                    <id>generate</id>
+                    <goals>
+                        <goal>generate</goal>
+                    </goals>
+                </execution>
+            </executions>
+        </plugin>
+    </plugins>
+</build>
+<repositories>
+    <repository>
+        <id>spring-releases</id>
+        <name>Spring Releases</name>
+        <url>https://repo.spring.io/release</url>
+        <snapshots>
+            <enabled>false</enabled>
+        </snapshots>
+    </repository>
+</repositories>
+<pluginRepositories>
+    <pluginRepository>
+        <id>spring-releases</id>
+        <name>Spring Releases</name>
+        <url>https://repo.spring.io/release</url>
+        <snapshots>
+            <enabled>false</enabled>
+        </snapshots>
+    </pluginRepository>
+</pluginRepositories>
```

Comment:
```suggestion
```

nit: I'm guessing you only need the `repository` for the aot dep.  

That said, this doesn't hurt, and any time you want to bump a Spring Boot version you want to include the `pluginRepository` anyway (to get the newer `spring-boot-maven-plugin`)

Reply:
I tried removing both and it fails with:

```
[ERROR] Plugin org.springframework.experimental:spring-aot-maven-plugin:0.10.3 or one of its dependencies could
 not be resolved: Could not find artifact org.springframework.experimental:spring-aot-maven-plugin:jar:0.10.3 in 
 central (https://repo.maven.apache.org/maven2) -> [Help 1]
```

If I add `<pluginRepositories>` only, it fails too:

```
[ERROR] Failed to execute goal on project demo: Could not resolve dependencies for project 
 com.example:demo:jar:0.0.1-SNAPSHOT: Could not find artifact 
 org.springframework.experimental:spring-native:jar:0.10.3 in central (https://repo.maven.apache.org/maven2) -> [Help 1]
```

So, it seems both are needed.

Reply:
Doh, sorry, I see it's a plugin 🤦 

Sorry for the noise